### PR TITLE
Have cache and origin run on Cache.Port and Origin.Port, respectively

### PR DESCRIPTION
**Pull request by bbockelm**
This deprecates the Xrootd.Port configuration and replaces it with Origin.Port and Cache.Port.  By separating out the configs, we have some hope of making the cache module part of the "federation in a box" since the cache will need to be a standalone Xrootd instance.

With this, Server.WebPort and Cache.Port/Origin.Port (as appropriate) are a hard requirement for the various modules to be enabled, requiring some unit tests to set these values.

Additionally, XRootD can be told to bind to "any" available TCP port.  If one of the port values is set to 0, then convert it to `any` in the xrootd configuration.  This mode is not yet useful because we need to feed back the selected port into the pelican configuration (and the only way to get the selected port is to scrape through the xrootd logs) -- however, it'll be invaluable for running multiple unit tests in parallel in the future.

### CODE REVIEW

**Comment by bbockelm** [comment]
Update on the PR (for @jhiemstrawisc who I added as reviewer):
- This now adds a simple logging filter to allow us to trigger a callback on the contents of logs
- With the filter, we can look for xrootd's log line indicating the port it starts up on.
- Splits out the launching of the listener from the web UI.  Allows us to start the listener first, get the random port it selected, and then update our internal configuration to match.
- Adds type safety across the `token_scope.TokenScope` usage.
- Migrates all origin issuer URLs to the web UI instead of XRootD.
- Changes a handful of tests to bind to random ports instead of fixed ones.

**Review by jhiemstrawisc — CHANGES_REQUESTED** [review]
A few small questions and minor requests for comments, otherwise LGTM.

**Comment by turetske** [comment]
This will cause issues with the broker_client.go changes you made. https://github.com/PelicanPlatform/pelican/blob/ebff302dbd1132a0e16b7b2470986787e58a4acb/origin_ui/broker_client.go#L65C1-L66C1 Uses Xrootd.Port. I suggest doing a grep for "Xrootd_Port" and "Xrootd.Port" after a rebase in order to make sure any deprecated uses are removed.

**Review by turetske — CHANGES_REQUESTED** [review]
This will cause issues with the broker_client.go changes you made. https://github.com/PelicanPlatform/pelican/blob/ebff302dbd1132a0e16b7b2470986787e58a4acb/origin_ui/broker_client.go#L65C1-L66C1 Uses Xrootd.Port. I suggest doing a grep for "Xrootd_Port" and "Xrootd.Port" after a rebase in order to make sure any deprecated uses are changed appropriately.

File: docs/parameters.yaml [hunk 1]
```diff
@@ -361,12 +361,17 @@ components: ["client"]
...
   The origin's configured URL, as reported to XRootD. This is the file transfer endpoint for the origin.
-
-  This Url must not have the same port number as the one in Server.ExternalWebUrl (if any) or Server.WebPort
 type: url
 default: https://${Server.Hostname}:${Xrootd.Port}
 components: ["origin"]
 ---
+name: Origin.Port
+description: >-
+  The TCP port to be used by the origin service for serving files
```
[the remark this answers is not in the document]
Comment (bbockelm) [reply docs/parameters.yaml]: Done!

File: config/logging.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+var (
+	globalFilters RegexpFilterHook
+)
+
+func (fh *RegexpFilterHook) Levels() []log.Level {
+	return log.AllLevels
+}
+
+func (fh *RegexpFilterHook) Fire(entry *log.Entry) (err error) {
```
[the remark this answers is not in the document]
Comment (bbockelm) [reply config/logging.go:59]: Done!

**Review by jhiemstrawisc — APPROVED** [review]
LGTM

**Review by turetske — APPROVED** [review]
LGTM!

**Pull request merged**